src/engine/helper/pka_helper.c

```diff
@@ -629,6 +837,143 @@ int pka_rsa_mod_exp_crt(pka_bignum_t  *bn_value,
     return rc;
 }
 
+int pka_ecc_bn_pt_mult(pka_bignum_t *bn_p,
+                       pka_bignum_t *bn_a,
+                       pka_bignum_t *bn_b,
+                       pka_bignum_t *bn_x,
+                       pka_bignum_t *bn_y,
+                       pka_bignum_t *bn_multiplier,
+                       pka_bignum_t *bn_xr,
+                       pka_bignum_t *bn_yr)
+{
+    pka_operand_t  *mltpr;
+    ecc_point_t     pt, *result;
+    ecc_curve_t     curve;
+    int             rc;
+
+    PKA_ASSERT(bn_p          != NULL);
+    PKA_ASSERT(bn_a          != NULL);
+    PKA_ASSERT(bn_b          != NULL);
+    PKA_ASSERT(bn_x          != NULL);
+    PKA_ASSERT(bn_y          != NULL);
+    PKA_ASSERT(bn_multiplier != NULL);
+    PKA_ASSERT(bn_xr         != NULL);
+    PKA_ASSERT(bn_yr         != NULL);
+
+    return_if_handle_invalid(tls_handle);
+
+    bignum_to_ecc_operand(bn_p, &(curve.p));
+    bignum_to_ecc_operand(bn_a, &(curve.a));
+    bignum_to_ecc_operand(bn_b, &(curve.b));
+    bignum_to_ecc_operand(bn_x, &(pt.x));
+    bignum_to_ecc_operand(bn_y, &(pt.y));
+    mltpr  = bignum_to_operand(bn_multiplier);
+
+    result = pka_do_ecc_pt_mult(tls_handle, &curve, &pt, mltpr);
```

Comment:
To be consistent with the rest of the function calls, I think it is better to declare pka_do_ecc_pt_mult as:

   int pka_do_ecc_pt_mult(..., ecc_point *result)

Reply:
As discussed, the current declaration is consistent with the rest of function calls. But I'll initialize the `rc` variable at the declaration time.